### PR TITLE
[Update] Simplify the dependencies

### LINKED ISSUE #156: Reduce the complexity of dependencies
## 🚀 Feature

Pare down the requirements (i.e., `requirements.txt`) to only include "top-level" packages (i.e. those that are actually used and are not dependencies of said packages). Furthermore, instead of pinning the versions of these packages, specify minimum versions instead.

### Motivation

Over-specifying dependencies in the requirements makes using the library out-of-the-box on systems where you don't have full control of the available dependencies much trickier. This is very common when trying to use public/enterprise computing clusters (e.g. Compute Canada). Furthermore, tying sub-dependencies to specific versions while leaving the versions of core dependencies flexible (e.g. PyTorch, NumPy, etc.) leads to mismatching dependency errors. This will become increasingly problematic as the core dependencies are updated.

I understand the desire to fix the versions of everything for reproducibility. However, when the versions sub-dependencies are fixed *but not* the versions of core dependencies (again, PyTorch etc.) fixing the versions of these sub-dependencies is redundant.

### Pitch

Already, core dependencies (e.g., PyTorch) are specified using minimum versions. I propose removing sub-dependencies from `requirements.txt` and specifying the remaining core dependencies as minimum versions (with the currently specified version as the new minimum).

The resulting requirements file would look like this:

```
matplotlib>=3.5.2
numpy
Pillow>=9.0.0
progress>=1.6
ruamel.yaml>=0.17.21
ruamel.yaml.clib>=0.2.6
scikit-image>=0.19.3
scikit-learn>=1.0.2
scipy>=1.10.0
tensorboard>=2.9.1
timm>=0.5.4
torch>=1.12.0
torchaudio>=0.12.0
torchvision>=0.13.0
tqdm>=4.64.0
transformers>=4.30.0
wandb
aim
```

I've already tested this to check that the training script still runs. It only requires one change to `vit.py`, changing the import path of `to_2tuple` to: `from timm.models.layers import to_2tuple`.

### Alternatives

None.

### Additional context

None.

### PR DESCRIPTION
**Pull request by adamtupper**
## What does this PR do?

Removes sub-dependencies from `requirements.txt` and specifies the remaining core dependencies as minimum versions (base on the current versions, where fixed).

Fixes #156 #140 

### Does your PR introduce any breaking changes? If yes, please list them.

None. The import paths from `timm` modules are fixed in `vit.py` (and the file is formatted according to PEP-8 style guides).

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you list all the **breaking changes** introduced by this pull request?

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

### CODE REVIEW

**Review by Hhhhhhao — APPROVED** [review]
The update seems do not break train loops thus approved. We might need to release a new package also for the recent updates.
